appliance_catalog/views.py

```diff
@@ -141,8 +143,24 @@ def get_appliance_template(request, pk):
     logger.info('Getting and displaying YAML template for appliance')
 
     appliance = Appliance.objects.filter(pk=pk).get()
-    return HttpResponse(appliance.template, content_type="text/yaml")
 
+    # send message to GA
+    if settings.GOOGLE_ANALYTICS_PROPERTY_ID:
+        if request.user and request.user.username:
+            cid = request.user.username
+        else:
+            cid = 'random-{}'.format(uuid.uuid1())
+        params = urllib.parse.urlencode({'v': 1,
+                                         'tid': settings.GOOGLE_ANALYTICS_PROPERTY_ID,
+                                         'cid': cid,
+                                         't': 'event',
+                                         'ec': 'heat_template',
+                                         'ea': 'download',
+                                         'el': '{}-{}'.format(pk, appliance.name),
+                                         'ev': 0})
+        conn = http_client.HTTPConnection('www.google-analytics.com')
+        conn.request('POST', '/collect', params)
```

Comment:
I'd feel more comfortable with this inside of a try/catch... otherwise this could break the appliance catalog.

Reply:
will add try/catch